### PR TITLE
Find epoch start slot for estimated time elapsed

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Many epochs, `solana epoch-info` only includes an estimate of time elapsed based on current slot durations, eg:
```
Epoch Completed Time: 1day 7h 7m 23s*/3days 13h 55m 11s (2days 6h 47m 48s remaining)

* estimated based on current slot durations
```
This is because the epoch start slot was skipped and we don't try any other slots.

#### Summary of Changes
Iterate `get_block_time` until we find the first not-skipped block (or until we try 16 slots), and use its block time plus the current average slot duration since the first slot in the epoch to make a better estimate of epoch time-elapsed.

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
This will slow down the command response when slots at the beginning of the epoch were skipped.
Is this worth doing? Realistically, this is fixing what is only a minor annoyance, possible only to me.

File: cli/src/cluster_query.rs [hunk 1]
```diff
@@ -1094,27 +1094,45 @@ pub fn process_get_epoch(rpc_client: &RpcClient, _config: &CliConfig) -> Process
...
+                    let (slots, secs) = samples.iter().fold((0, 0), |(slots, secs), sample| {
+                        (slots + sample.num_slots, secs + sample.sample_period_secs)
+                    });
+                    (secs as u64).saturating_mul(1000).checked_div(slots)
+                })
+                .unwrap_or(clock::DEFAULT_MS_PER_SLOT);
+            let mut start_block_time = None;
+            let epoch_expected_start_slot = epoch_info.absolute_slot - epoch_info.slot_index;
+            let mut epoch_start_slot = epoch_expected_start_slot.saturating_sub(1);
+            while start_block_time.is_none() && epoch_start_slot <= epoch_expected_start_slot + 16 {
```
Comment (t-nelson) [inline cli/src/cluster_query.rs:1120]: any reason not to cheat and `getBlocksWithLimit(epoch_expected_start_slot, 1)` our way to victory?

Comment (CriesofCarrots) [reply cli/src/cluster_query.rs:1120]: No indeed! That would be much better. Will rewrite.

Comment (CriesofCarrots) [reply cli/src/cluster_query.rs:1120]: Well, I suppose the one possible argument against it is that get_blocks_with_limit doesn't return the block_time. So there's an extra RPC call needed in the case where the epoch doesn't start with a skipped slot. I still think it's better, though.

Comment (CriesofCarrots) [reply cli/src/cluster_query.rs]: @t-nelson , updated. How does that look?
I considered checking the block time of the expected epoch start slot first, and only pressing on if None, but the early block time requests are taking quite a long time against the foundation rpc cluster atm (I'm not totally sure why; need to take a deeper look) so I'm reluctant to add an extra one.

**Review by t-nelson — APPROVED** [review]
lgtm!

**Pull request merged**